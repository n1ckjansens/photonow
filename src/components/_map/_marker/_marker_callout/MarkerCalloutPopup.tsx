import React from 'react'
import MarkerCalloutLayout from './MarkerCalloutLayout'

const MarkerCalloutPopup: React.FC = () => {
	return (
		<div className="callout__container">
			<MarkerCalloutLayout />
		</div>
	)
}

export default MarkerCalloutPopup
