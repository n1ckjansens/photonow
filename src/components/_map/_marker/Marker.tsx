import React from 'react'
import MarkerCalloutPopup from './_marker_callout/MarkerCalloutPopup'

//Types if props component recieve
interface Props {
	showCallout : boolean,
	lat: number, 
	lng: number
}

//Main functional component
const Marker: React.FC<Props> = ({ showCallout }) => {
	return (
		<div className="marker__container">
			{showCallout && <MarkerCalloutPopup />}
			<div className="pin bounce">
				<div className="user__avatar">
					<img
						src="https://sun1-47.userapi.com/impg/c856124/v856124902/1fba84/39E3gZWT5qk.jpg?size=200x0&quality=90&crop=416,0,1334,1334&sign=8e096f2c133c95e3720669fa1b84b1f5&ava=1"
						alt="user__avatar"
					/>
				</div>
			</div>
			<div className="pulse" />
		</div>
	)
}

export default Marker
