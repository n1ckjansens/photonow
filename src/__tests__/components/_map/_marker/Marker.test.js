import React from 'react'
import { shallow } from 'enzyme'
import Marker from '../../../../components/_map/_marker/Marker'
import MarkerCalloutPopup from '../../../../components/_map/_marker/_marker_callout/MarkerCalloutPopup'

//Function to render component into test
const setUp = (props) => shallow(<Marker {...props} />)

describe('Marker component test', () => {
	let component, componentWithoutCallout

	//Rendering component before each test
	beforeEach(() => {
		componentWithoutCallout = setUp({ showCallout: false })
		component = setUp({ showCallout: true })
	})

	//Testing for successfull rendering of map through .map__container wrapper
	it('Should render successfully and contain .marker__container wrapper', () => {
		const marker__container = component.find('.marker__container')
		expect(marker__container.length).toBe(1)
	})

	//Testing that callout doesn't shows when showCallout prop is false
	it('Should non contain callout component when showCallout prop is false', () => {
		const callout = componentWithoutCallout.find(MarkerCalloutPopup)
		expect(callout.length).toBe(0)
	})

	//Testing if calloutPopup shows when showCallout prop is true and window.innerWidth prop is grater than 750px
	it('Should contain MarkerCalloutPopup component when showCallout prop is true and window.innerWidth prop is grater than 750px', () => {
		//setting window width
		window.innerWidth = 800
		//rendering component od this width
		const coponentWithCallout = setUp({ showCallout: true })
		//finding callout component
		const callout = coponentWithCallout.find(MarkerCalloutPopup)
		expect(callout.length).toBe(1)
	})

	//Testing component matches previos snapshots
	it('Should match snapshot', () => {
		expect(component).toMatchSnapshot()
	})
})
