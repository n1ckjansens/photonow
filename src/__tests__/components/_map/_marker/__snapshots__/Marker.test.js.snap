// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker component test Should match snapshot 1`] = `
<div
  className="marker__container"
>
  <MarkerCalloutPopup />
  <div
    className="pin bounce"
  >
    <div
      className="user__avatar"
    >
      <img
        alt="user__avatar"
        src="https://sun1-47.userapi.com/impg/c856124/v856124902/1fba84/39E3gZWT5qk.jpg?size=200x0&quality=90&crop=416,0,1334,1334&sign=8e096f2c133c95e3720669fa1b84b1f5&ava=1"
      />
    </div>
  </div>
  <div
    className="pulse"
  />
</div>
`;
